index.js

```diff
@@ -0,0 +1,219 @@
+const URL = 'https://ada-backtrek-api.herokuapp.com/trips/';
+
+// Report helpers
+const reportStatus = (message) => {
+  $('#status-message').html(message);
+}
+
+const reportError = (message, errors) => {
+  let content = `<p>${message}</p><ul>`;
+  for (const field in errors) {
+    for (const problem of errors[field]) {
+      content += `<li>${field}: ${problem}</li>`;
+    }
+  }
+  content += "</ul>";
+  reportStatus(content);
+};
+
+//LOAD all/specific trips based on url params
+const getTrips = (url) => {
+  const tripList = $('#trip-list');
+  tripList.show();
+  tripList.empty();
+
+  reportStatus('Loading trips, please wait...');
+
+  axios.get(url)
+    .then((response) => {
+      let header = $(`<h4>All Trips</h4>`)
+      tripList.append(header);
+
+      response.data.forEach((trip) => {
+        let item = $(`<li>${trip.name}</li>`).attr('id', `${trip.id}`);
+        tripList.append(item);
+      });
+
+      reportStatus(`Successfully loaded ${response.data.length} trips`)
+    })
+    .catch((error) => {
+      console.log(error);
+      reportStatus(`Error: ${error.message}`);
+    });
+};
+
+// GET ALL TRIPS
+const loadTrips = () => {
+  getTrips(URL)
+}
+// GET ASIA
+const asiaTrips = () => {
+  let url = (URL + '/continent?query=Asia')
+  getTrips(url)
+}
+// GET AFRICA
+const africaTrips = () => {
+  let url = (URL + '/continent?query=Africa')
+  getTrips(url)
+}
+// GET Antartica
+const antarticaTrips = () => {
+  let url = (URL + '/continent?query=Antartica')
+  getTrips(url)
+}
+// GET Australasia
+const australasiaTrips = () => {
+  let url = (URL + '/continent?query=Australasia')
+  getTrips(url)
+}
+// GET EUROPE
+const europeTrips = () => {
+  let url = (URL + '/continent?query=Europe')
+  getTrips(url)
+}
+// GET North America
+const nAmericaTrips = () => {
+  let url = (URL + '/continent?query=North%20America')
+  getTrips(url)
+}
+// GET South America
+const sAmericaTrips = () => {
+  let url = (URL + '/continent?query=South%20America')
+  getTrips(url)
+}
+
+// GET details for single trip
+const getTripDetails = function getTripDetails(id) {
+
+  let tripDetail = $('#trip-detail-section');
+  tripDetail.empty();
+
+  axios.get(URL + id)
+    .then((response) => {
+      let data = response.data;
+      let name = $(`<h4><strong>Name:</strong> ${data.name}</h4>`).addClass(`${id}`);
```

Comment:
It might make more sense to set the `id` attr on this element instead of adding a class, because this value represents a unique identifier rather than a styling class?